scripts/pre_commit_linter.py

```diff
@@ -1672,25 +1697,26 @@ def _check_docstrings(self):
                                 if last_char_is_invalid and (
                                         no_word_is_present_in_excluded_phrases):
                                     failed = True
-                                    print '%s --> Line %s: %s' % (
+                                    print('%s --> Line %s: %s' % (
                                         filepath, line_num,
-                                        missing_period_message)
-                                    print ''
+                                        missing_period_message))
+                                    print('')
 
                         # Case 2: line contains some words before """. """
                         # should shift to next line.
                         elif not any(word in line for word in EXCLUDED_PHRASES):
                             failed = True
-                            print '%s --> Line %s: %s' % (
+                            print('%s --> Line %s: %s' % (
                                 filepath, line_num + 1,
-                                multiline_docstring_message)
-                            print ''
+                                multiline_docstring_message))
+                            print('')
 
                         is_docstring = False
 
             docstring_checker = docstrings_checker.ASTDocStringChecker()
             for filepath in files_to_check:
-                ast_file = ast.walk(ast.parse(FileCache.read(filepath)))
+                ast_file = ast.walk(
+                    ast.parse(FileCache.read(filepath).encode('utf-8')))
```

Comment:
Likewise? Maybe we should move the encoding/decoding into `FileCache` instead so contributors don't need to worry about it?

Reply:
done